File: src/main/java/cn/bromine0x23/plantuml/controllers/render/TxtController.java
package cn.bromine0x23.plantuml.controllers.render;

import net.sourceforge.plantuml.FileFormat;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

/**
 * 渲染PlantUML为文本格式
 *
 * @author <a href="mailto:bromine0x23@163.com">Bromine0x23</a>
 */
@RestController
@RequestMapping({"/txt"})
public class TxtController extends AbstractRenderController {

	private static final MediaType CONTENT_TYPE = MediaType.parseMediaType(MediaType.TEXT_PLAIN_VALUE + ";charset=UTF-8");

	@Override
	protected FileFormat getFileFormat() {
		return FileFormat.UTXT;
	}

	@Override
	protected MediaType getContentType() {
		return CONTENT_TYPE;
	}
}

File: src/main/kotlin/cn/bromine0x23/plantuml/controllers/TxtController.kt
package cn.bromine0x23.plantuml.controllers

import cn.bromine0x23.plantuml.services.RenderService
import cn.bromine0x23.plantuml.services.UmlService
import net.sourceforge.plantuml.FileFormat
import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/txt")
class TxtController(
	umlService: UmlService,
	renderService: RenderService
): AbstractRenderController(umlService, renderService) {

	companion object {
		@JvmStatic
		private val CONTENT_TYPE = MediaType.valueOf(MediaType.TEXT_PLAIN_VALUE + ";charset=UTF-8")
	}

	override val format: FileFormat
		get() = FileFormat.UTXT

	override val contentType: MediaType
		get() = CONTENT_TYPE
}